es6/timeline.js

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import {connect} from 'react-redux'
+import {ExpandableSection} from 'monadical-react-components'
+
+const SOURCE = "https://github.com/Monadical-SAS/redux-time/blob/master/es6/timeline.js"
+const COMPONENT_HEIGHT = 320
+
+const page_start_time = Date.now()
+
+const offset = (time) =>
+    time - page_start_time
+
+
+const AnimRow = (anim, idx, scale) => {
+    const {type, start_time, end_time} = anim
+
+    const style = {
+        position: 'absolute',
+        top: (idx * 25) % (COMPONENT_HEIGHT - 70),
+        left: offset(start_time)/scale,
+        height: 20,
+        width: end_time !== Infinity ?
+                (end_time - start_time)/scale
+                : 10,
+        backgroundColor: type.includes('BECOME') ? 'gray' : 'red',
+        border: 'solid 1px black',
+        overflow: 'hidden',
+    }
+
+    return <div className="anim" style={style}>
+            {type}
+            <br/>
+            <div className="anim_details">
+                Start time: {`${anim.start_time}`}<br/>
+                End time: {`${anim.end_time}`}<br/>
+                Start state: {JSON.stringify(anim.start_state)}<br/>
+                End state: {JSON.stringify(anim.end_state)}<br/>
+                Curve: {`${anim.curve}`}<br/>
+            </div>
+        </div>
+}
+
+const CurrentFrame = ({warped_time, scale}) => {
+
+    const style = {
+        position: 'absolute',
+        top: COMPONENT_HEIGHT - 70,
+        left: (offset(warped_time) / scale) - 1,
+        height: 20,
+        color: 'blue',
+    }
+
+    return <div style={style}>|</div>
+}
+
+class TimelineComponent extends React.Component {
```

Comment:
I think you need to pull from time-slider branch

Reply:
Planning on merging that into dev afterwards (separately)